src/platform/backends/libvirt/libvirt_virtual_machine.cpp

```diff
@@ -332,6 +342,16 @@ int mp::LibVirtVirtualMachine::ssh_port()
     return 22;
 }
 
+void mp::LibVirtVirtualMachine::ensure_vm_is_running()
+{
+    std::lock_guard<decltype(state_mutex)> lock{state_mutex};
+    if (domain_state_for(domain.get()) != VirtualMachine::State::running)
+    {
+        state_wait.notify_all();
```

Comment:
What I'm trying to prevent is the daemon continuing it's operation until it's safe to so.  The situation I'm trying to avoid is a `multipass delete -p` on a starting instance and the daemon destroying the corresponding *VirtualMachine object while it's starting since the starting check is in another thread.  This is a synchronization between the daemon thread and the async_wait_until_ssh_up() thread.